MiniApp/Classes/Protocols/MiniAppMessageProtocol.swift

```diff
@@ -19,6 +19,18 @@ public protocol MiniAppMessageProtocol: class {
     func shareContent(info: MiniAppShareContent, completionHandler: @escaping (Result<MASDKProtocolResponse, Error>) -> Void)
 }
 
+/**
+ Public Protocol that will be used by the Mini App to communicate
+ with the Native implementation for User profile related retrieval
+*/
+public protocol MiniAppUserInfoProtocol: class {
+    /// Interface that is used to retrieve the user name from the User Profile
+    func getUserName() -> String
```

Comment:
I think I misunderstood something about how these protocols are working on iOS. When we add new functions to this protocol in the future, then the Host App will be required to implement the new functions or else their build will fail, right?

I had assumed this wasn't the case because in `RealMiniApp` we are implementing default handlers for the functions. However, these default handlers only apply if the Host App doesn't provide `MiniAppMessageBridgeProtocol` or `MiniAppUserInfoProtocol` at all, correct?

This is a problem because it means we will introduce breaking changes every time we want to add new methods to the protocol. So we need to think of how we can add new functions without the Host App being required to implement them. 

On the Android side, we switched from using `abstract` functions to just normal `open` functions which have either a default implementation or throw an error when the Host App hasn't overridden them. Is something similar possible for iOS? You lose the benefit of requiring a certain implementation, but maybe that's okay because the message interfaces are intended to be optional anyways.

Any ideas @leorjoseph @Climbatize ?

Reply:
Yeah unfortunately sift does not allow optional methods. There are two ways to solve this, either by annotating the protocol with @objc and use exclusively Objc compliant objects into it, this seems the case here. This way you can make functions optional and have a way to override their implementation if they are not implemented. Either by making subprotocols to keeps swift compliance, you can take a look at my current #151 PR to have an example.

Another approach would be to subclass UIViewController, UItableViewConreoller, UICollectionViewControler, etc. as MiniAppViewController, MiniAppViewController, MiniAppTableViewController, etc. to implement a base behavior, and then ask host developers to rely on these classes instead of base UIKit classes. This is the smartest approach to have abstracts methods, but it can quickly become a nightmare for developers of existing apps who just want to add support for a MiniApp and already have subclasses of UIKit components.

Reply:
I guess Protocol extensions can overcome this problem. Host app can conform to the existing protocols(which we have already) and we can create a variable that conforms to multiple/other protocols. I will share the example project in Slack and we can take the decision forward on this.

Reply:
@Climbatize , We can make use of Protocol extensions to avoid these situations. I have updated the code, have a look.